portfolio/src/main/webapp/slideshow_script.js

```diff
@@ -0,0 +1,55 @@
+//sliding
+var slideIndex = 1;
+var timeout;
+showSlide(slideIndex); //initial state
+
+function showSlide(index){
+    var slides = document.getElementsByClassName("slide");
+    var noSlides = slides.length
+    console.log(noSlides);
+    if(index>noSlides || index<1 ){
+        index=(index%noSlides+noSlides)%noSlides+1;
```

Comment:
Micro nit: Since the name of this function is `showSlide(index)`, its responsibility should only be to show a slide at `index`.  This computation would fit better in a function `incrementSlideIndexBy(n)`.

Reply:
I extracted this part into the incrementSlideIndex(n) function